Indicators/KlingerVolumeOscillator.cs

```diff
@@ -0,0 +1,166 @@
+        /// </summary>
+        /// <param name="name">The name of the indicator.</param>
+        /// <param name="fastPeriod">The fast EMA period.</param>
+        /// <param name="slowPeriod">The slow EMA period.</param>
+        public KlingerVolumeOscillator(string name, int fastPeriod, int slowPeriod)
+            : base(name)
+        {
+            _fastPeriod = fastPeriod;
+            _slowPeriod = slowPeriod;
+
+            _fastEma = new ExponentialMovingAverage(name + "_FastEma", fastPeriod);
+            _slowEma = new ExponentialMovingAverage(name + "_SlowEma", slowPeriod);
+
+            _hasPrevBar = false;
+            _previousHlc3 = 0m;
+            _previousDm = 0m;
+            _previousCm = 0m;
+            _previousTrend = 0;
+
+            WarmUpPeriod = Math.Max(fastPeriod, slowPeriod) + 2;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="KlingerVolumeOscillator"/> class with specified fast, slow periods.
+        /// </summary>
+        /// <param name="fastPeriod">The fast EMA period.</param>
+        /// <param name="slowPeriod">The slow EMA period.</param>
+        public KlingerVolumeOscillator(int fastPeriod, int slowPeriod)
+            : this($"KVO({fastPeriod},{slowPeriod})", fastPeriod, slowPeriod)
+        {
+        }
+
+        /// <summary>
+        /// Computes the next value of the Klinger Volume Oscillator based on the input data point.
+        /// </summary>
+        protected override decimal ComputeNextValue(TradeBar bar)
+        {
+            // daily movement
+            var dm = bar.High - bar.Low;
+            var hlc3 = (bar.High + bar.Low + bar.Close) / 3m;
+
+            if (!_hasPrevBar)
+            {
+                // First bar
+                _previousDm = dm;
+                _previousHlc3 = hlc3;
+                _previousTrend = 0;
+                _hasPrevBar = true;
+                return 0m;
+            }
+
+            // trend direction
+            var currentTrend = hlc3 > _previousHlc3 ? 1 : -1;
+
+            if (_previousTrend == 0)
+            {
+                // Second bar
+                _previousTrend = currentTrend;
+                _previousDm = dm;
+                _previousHlc3 = hlc3;
+                return 0m;
+            }
+
+            // Cumulative measurement (cm): Accumulates daily price range (High - Low) based on trend continuity
+            var cm = 0m;
+            var cum = 0m;
+            if (_previousCm == 0m || currentTrend != _previousTrend)
+            {
+                // Start new flow accumulation with the previous daily movement (dm)
+                cum = _previousDm;
+            }
+            else
+            {
+                // Continue flow accumulation in the same trend direction
+                cum = _previousCm;
+            }
+            cm = cum + dm;
```

Comment:
Generically we try to expand all variable names/private fields so they can easily be understood, for example `cm` -> `cumulative`, `dm` -> etc